torch/csrc/jit/codegen/cuda/test/test_gpu_compute_with.cpp

```diff
@@ -0,0 +1,451 @@
+#include <torch/csrc/jit/codegen/cuda/test/test_utils.h>
+#include <torch/csrc/jit/codegen/cuda/transform_replay.h>
+#include <torch/csrc/jit/codegen/cuda/transform_rfactor.h>
+
+#include <test/cpp/jit/test_utils.h>
+#include <torch/csrc/jit/api/function_impl.h>
+#include <torch/csrc/jit/codegen/cuda/parser.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/torch.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/Exceptions.h>
+#include <c10/cuda/CUDAStream.h>
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <thread>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+using namespace torch::jit::fuser::cuda;
+using namespace at::indexing;
+
+TEST_F(NVFuserTest, FusionComputeWith1_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeContigTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = set(tv0);
+  auto tv2 = sum(tv1, {1});
+  auto tv3 = broadcast(tv2, {false, true});
+  auto tv4 = add(tv3, tv1);
+  fusion.addOutput(tv4);
+
+  inlineMost();
+
+  tv1->computeWith(tv2, -1);
+
+  tv1->setMemoryType(MemoryType::Shared);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  at::Tensor t0 = at::randn({10, 11}, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {t0});
+  auto cg_outputs = fe.runFusion({t0});
+
+  auto ref = t0.sum({1}).unsqueeze(-1) + t0;
+
+  testValidate(&fusion, cg_outputs, {t0}, {ref}, __LINE__, __FILE__);
+}
+
+// StoreAt with 1D softmax
+TEST_F(NVFuserTest, FusionComputeWith2_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  const int vec = 4;
+  const int tidx = 128;
+  const int dimx = 1000;
+
+  auto input_tv0 = makeContigTensor(1);
+  fusion.addInput(input_tv0);
+
+  auto exp_tv1 = unaryOp(UnaryOpType::Exp, input_tv0);
+  auto sum_exp_tv2 = sum(exp_tv1, {-1});
+  auto bcast_sum_tv3 = broadcast(sum_exp_tv2, {true});
+
+  auto exp_tv1_copy = unaryOp(UnaryOpType::Exp, input_tv0);
+
+  auto output_tv4 = div(exp_tv1_copy, bcast_sum_tv3);
+
+  fusion.addOutput(output_tv4);
+
+  auto input_tv0_cache = input_tv0->cacheAfter();
+
+  input_tv0->split(-1, vec);
+  input_tv0->split(-2, tidx);
+  MaxRootDomainInfoSpanningTree tree(input_tv0);
+  TransformPropagatorWithCheck tp(input_tv0);
+  tree.traverse(&tp);
+
+  auto sum_exp_rf_tv5 = sum_exp_tv2->rFactor({-1});
+
+  inlineMost();
+  input_tv0_cache->computeWith(exp_tv1, -1);
+
+  input_tv0_cache->axis(0)->parallelize(ParallelType::BIDx);
+  input_tv0_cache->axis(1)->parallelize(ParallelType::TIDx);
+  scheduler_utils::parallelizeAllLike(input_tv0_cache);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({dimx}, options);
+  at::Tensor cg_output = at::empty({dimx}, options);
+  at::Tensor t3_output = at::empty_like(cg_output, options);
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, {t0});
+  fe.runFusion({t0}, {cg_output});
```

Comment:
Why not just
```
auto cg_outputs = fe.runFusion({t0});
```
?

Reply:
Not sure. Just copied an existing old softmax test. Updated.